docs/contributing/using-instrumenter-api.md

```diff
@@ -1 +1,468 @@
+The `start()` method begins gathering telemetry about the instrumented library function that's being
+invoked. It starts the `Span` and begins recording the metrics (if any are registered in
+the used `Instrumenter` instance).
+
+The `start()` method accepts the current OpenTelemetry `Context` and the instrumented
+library `REQUEST` type, and returns the new OpenTelemetry `Context` that should be made current
+until the instrumented operation ends. Consider the following example:
+
+```java
+Response decoratedMethod(Request request) {
+  // ...
+
+  Context context = instrumenter.start(parentContext, request);
+  try (Scope scope = context.makeCurrent()) {
+    // ...
+  }
+}
+```
+
+The newly started `context` is made current, and inside its `scope` the actual library method is
+called. After that the instrumented operation ends.
+
+## End an instrumented operation using `end()`
+
+The `end()` method is called when the instrumented operation finished. It is of extreme importance
+for this method to be always called after `start()`. Calling `start()` without later `end()`
+may result in inaccurate or wrong telemetry and context leaks. The `end()` method ends the current
+span and finishes recording the metrics (if any are registered in the `Instrumenter` instance).
+
+The method accepts several arguments:
+
+* The current OpenTelemetry `Context` - that is, the one that was returned by the `start()` method.
+* The `REQUEST` instance that has started the processing.
+* Optionally, the `RESPONSE` instance that ends the processing - it may be `null` in case it was not
+  received or an error has occurred.
+* Optionally, a `Throwable` error that was thrown during the processing.
+
+Consider the following example:
+
+```java
+Response decoratedMethod(Request request) {
+  // ...
+
+  Context context = instrumenter.start(parentContext, request);
+  try (Scope scope = context.makeCurrent()) {
+    Response response = actualMethod(request);
+    instrumenter.end(context, request, response, null);
+    return response;
+  } catch (Throwable error) {
+    instrumenter.end(context, request, null, error);
+    throw error;
+  }
+}
+```
+
+In the code sample the `context` returned by the `start()` method is passed along to the `end()`
+method. The `end()` method is always called, regardless of the outcome of the
+decorated `actualMethod()` - be it a valid response or an error.
+
+## Constructing a new `Instrumenter` using an `InstrumenterBuilder`
+
+An `Instrumenter` can be obtained by calling its static `builder()` method and using the
+returned `InstrumenterBuilder` to configure captured telemetry and apply customizations to the new
+instance. The `builder()` method accepts three arguments:
+
+* An `OpenTelemetry` instance, which will be used to obtain the `Tracer` and `Meter` objects.
+* The instrumentation name - it denotes the _instrumentation_ library name, not the _instrumented_
```

Comment:
Add a link to the spec document explaining the difference?

Reply:
Good idea, done.